neuralpp/symbolic/function.py

```diff
@@ -0,0 +1,46 @@
+from typing import Callable
+from abc import ABC, abstractmethod
+
+
+class Function(ABC):
+    """
+    The motivation of having a dedicated Function class is to support `built-in functions`.
```

Comment:
I am not quite seeing the argument here, or the need for Function(). If we use `BasicConstant(operator.add)` then we can recognize `operator.add` as a supported function and do the same things we could do with `Add()`, but without having to define `Function` and `Add`, isn't it so?

Reply:
I think my general objection that i feel it's a bit messy to compare "opeartors". Specifically, in Python, the following test would pass:
```
class OperatorTestClass:
    __match_args__ = ("op")

    def __init__(self, op):
        self._operator = operator

    @property
    def op(self):
        return self._operator


def test_operator_matching():
    e = OperatorTestClass(operator.add)
    e2 = OperatorTestClass(operator.add)
    assert operator.add == operator.add
    assert e.op != operator.add  # a bit counterintuitive here
    assert e.op == e.op
    assert e.op == e2.op

    match e:
        case OperatorTestClass(op=operator.add):
            assert False  # not reachable
        case OperatorTestClass(op=op):  # property is also accepted as __match_args__
            assert op != operator.add
            assert op == e2.op
```

Reply:
I think this is just a bug in your test code . Your `__init__` method receives argument `op` but uses `operator` instead.

Reply:
Yeah you're right.

Reply:
removed `function.py`